client/landing/gutenboarding/hooks/use-typer.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useInterval } from 'lib/interval/use-interval';
+
+/**
+ * A React hook that returns typing-machine animated strings
+ *
+ * @param {Array<string>} texts An array of strings you want to create the typing effect for
+ * @param {boolean?} enabled Whether the animation is enabled
+ * @param {number?} delayBetweenCharacters The typing delay time between two charactors. Default is 120ms
+ * @param {number?} delayBetweenWords The typing delay time between two words. Default is 1200ms
+ */
+export default function useTyper(
+	texts: Array< string >,
+	enabled = true,
+	delayBetweenCharacters = 120,
+	delayBetweenWords = 1200
```

Comment:
What do you think about all the parameters listed vs. putting some, like these configuration values, into an object?

Reply:
I felt this was a little over-complicated for its value but the hook API looks much better now. Adopted.